control/Block.js

```diff
@@ -379,14 +380,110 @@ async function setupTagRules(pid, tags, dstTag, dstType, allow = false, destroy
         }
       }
 
-      log.info('Finish', destroy ? 'destroying' : 'creating', 'block environment for', tags || "null", dstTag);
+      log.info('Finish', destroy ? 'destroying' : 'creating', 'block environment for', pid || "null", dstTag);
 
     } catch(err) {
       log.error('Error when setup tag blocking env', err);
     }
   }
 }
 
+async function setupIntfsRules(pid, intfs, dstTag, dstType, allow = false, destroy = false, destroyDstCache = true) {
+  if (_.isEmpty(intfs)) {
+    return;
+  }
+
+  const NetworkProfileManager = require('../net2/NetworkProfileManager.js')
+  for (let index = 0; index < intfs.length; index++) {
+    if (!NetworkProfileManager.getNetworkProfile(intfs[index])) {
```

Comment:
You can use NetworkProfile.ensureCreateEnforcementEnv(uuid) to ensure the ip set is created even if the network profile does not exist at the moment. Since network may be restored from config history, the policy on the restored network should take effect at once.